src/main/java/de/fraunhofer/aisec/cpg/graph/DeclaredReferenceExpression.java

```diff
@@ -43,63 +44,82 @@
 public class DeclaredReferenceExpression extends Expression implements TypeListener {
 
   /** The {@link ValueDeclaration}s this expression might refer to. */
-  private Set<ValueDeclaration> refersTo = new HashSet<>();
+  private Set<Declaration> refersTo = new HashSet<>();
 
   /**
    * Is this reference used for writing data instead of just reading it? Determines dataflow
    * direction
    */
   private AccessValues access = AccessValues.READ;
 
-  public Set<ValueDeclaration> getRefersTo() {
+  private boolean staticAccess = false;
+
+  public boolean isStaticAccess() {
+    return staticAccess;
+  }
+
+  public void setStaticAccess(boolean staticAccess) {
+    this.staticAccess = staticAccess;
+  }
+
+  public Set<Declaration> getRefersTo() {
     return refersTo;
   }
 
   public AccessValues getAccess() {
     return access;
   }
 
-  public void setRefersTo(@NonNull ValueDeclaration refersTo) {
-    HashSet<ValueDeclaration> n = new HashSet<>();
+  public void setRefersTo(@Nullable Declaration refersTo) {
+    if (refersTo == null) {
+      return;
+    }
+    HashSet<Declaration> n = new HashSet<>();
     n.add(refersTo);
     setRefersTo(n);
   }
 
-  public void setRefersTo(@NonNull Set<ValueDeclaration> refersTo) {
-    this.refersTo.forEach(
-        r -> {
-          if (access == AccessValues.WRITE) {
-            this.removeNextDFG(r);
-          } else if (access == AccessValues.READ) {
-            this.removePrevDFG(r);
-          } else {
-            this.removeNextDFG(r);
-            this.removePrevDFG(r);
-          }
-          r.unregisterTypeListener(this);
-          if (r instanceof TypeListener) {
-            this.unregisterTypeListener((TypeListener) r);
-          }
-        });
+  public void setRefersTo(@NonNull Set<Declaration> refersTo) {
+    this.refersTo.stream()
```

Comment:
So we are filtering only for ValueDeclarations anyway? So I do not see the need for settings the set to declarations.